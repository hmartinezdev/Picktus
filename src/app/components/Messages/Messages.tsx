import React, { Component } from 'react';
import { Transition, TransitionGroup } from 'react-transition-group';
import { IMessageHandlerProps, IMessageHandlerState } from './Messages.type';
import Notification from './Notification';

class MessageHandler extends Component<IMessageHandlerProps, IMessageHandlerState> {
  public static defaultProps = {
    notifications: [],
    snackbars: [],
  };

  public render(): React.ReactElement<MessageHandler> {
    const { notifications } = this.props;
    return (
      <div className="container">
        <TransitionGroup className="notifications" appear={true}>
          {notifications.map((value, key) => {
            if (key === 0) {
              return (
                <Transition key={value.id} timeout={700} mountOnEnter={true} unmountOnExit={true}>
                  {(status) => (
                    <div className={`notification__container notification__container--${status}`}>
                      <Notification text={value.text} level={value.level} />
                    </div>
                  )}
                </Transition>
              );
            }
          })}
        </TransitionGroup>
        <style jsx>{`
          .container {
          }

          .notification__container {
            position: absolute;
            top: 4vh;
            right: -3px;
            transition: all 700ms ease-out;
            transform: translateX(120%);
            z-index: 1000;
          }

          .notification__container--entering,
          .notification__container--entered {
            transform: translateX(0%);
          }

          .notification__container--exiting,
          .notification__container--exited {
            transform: translateX(120%);
          }
        `}</style>
      </div>
    );
  }
}

export default MessageHandler;
