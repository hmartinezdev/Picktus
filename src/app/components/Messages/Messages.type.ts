import { IPicktusError, IPicktusMessage } from '@store/reducers/message';

export interface IReduxStateProps {
  snackbars: Array<IPicktusError | IPicktusMessage>;
}

export interface IMessageHandlerState {
  snackbar: IPicktusError | IPicktusMessage | null;
}

export type IMessageHandlerProps = IReduxStateProps;
