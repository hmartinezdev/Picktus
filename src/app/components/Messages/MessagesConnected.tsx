import { IReduxState } from '@store/reducers';
import { connect } from 'react-redux';

import Messages from './Messages';
import { IReduxStateProps } from './Messages.type';

const mapStateToProps = (state: IReduxState): IReduxStateProps => {
  return {
    snackbars: state.message.snackbars,
  };
};

export default connect(mapStateToProps)(Messages);
