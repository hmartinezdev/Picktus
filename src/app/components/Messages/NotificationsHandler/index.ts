import NotificationsHandler from './NotificationsHandler';

export default NotificationsHandler;
