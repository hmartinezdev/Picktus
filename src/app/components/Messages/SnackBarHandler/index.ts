import SnackBarHandlerConnected from './SnackBarHandlerConnected';

export default SnackBarHandlerConnected;
