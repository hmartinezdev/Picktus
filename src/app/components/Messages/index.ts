import MessagesConnected from './MessagesConnected';

export default MessagesConnected;
