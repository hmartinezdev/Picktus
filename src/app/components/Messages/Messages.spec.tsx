import { shallow } from 'enzyme';
import React from 'react';
import Messages from './Messages';

let wrapper;
const baseProps = {
  notifications: [],
  snackbars: [],
};

const setup = (props = baseProps) => shallow(<Messages {...props} />);

describe('<MessageHandler />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
