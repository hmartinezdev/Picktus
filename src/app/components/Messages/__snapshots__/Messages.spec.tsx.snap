// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageHandler /> should render properly 1`] = `
<div
  className="container"
>
  <ConnectFunction />
  <ConnectFunction />
</div>
`;
